### ISSUE TITLE
Regression in \verbatim & \autodoc:codeblock handling of blank lines

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/sile#2128]
As reported by @chrisvest [on mastodon](https://mastodon.social/@chrisvest/113263045558380660) the SILE manual has an issue with whitespace handling in code samples. We've lost all blank line to paragraph transformations in code samples. These days in the manual that's with `\autodoc:codeblock`, but the mechanism should be essentially the same as `\verbatium`. I've tried to get an MWE to bisect but I can't find an MWE that works/doesn't work in a reliable way. In fact at the moment even in versions where the manual appears to be correct my MWE's do *not* work.

Paging back through release manuals, we apparently completely broke all the code sample rendering in v0.14.10. That would make sense if you just looked at the fact that we used all now formatting macros for the whole manual starting in that release, but even restoring the v0.14.9 code for rendering code blocks or building a quick check with `\verbatium` still fails even in v0.14.9. And indeed all the way back to v0.11.0 at least.

Here is the v0.14.9 manual:

![image](https://github.com/user-attachments/assets/38f4e678-c212-467a-9cca-c4a255133a81)

One release later in v0.14.10 the `parbreak` is gone:

![image](https://github.com/user-attachments/assets/7edc64da-20c7-4dd6-ab79-85ae915e393a)

And here it is today in v0.15.5:

![image](https://github.com/user-attachments/assets/d58634df-4cea-4d7d-9bd6-6b1003d6e9e8)

The source is still to this day like I would expect:

```sile
\begin[type=autodoc:codeblock]{raw}
Paragraph one.

Paragraph two.
This is not paragraph three.

This is paragraph three.
\end{raw}
```

Any ideas @Omikhleia?

**Comment by Omikhleia** [comment]
> Any ideas @Omikhleia?

Hmmm. It if ever worked and got broken at one point, it might even have occurred early (I remember having added explicit spaces in my own verbatim examples some long time ago...). 

I also mentioned the issue here in a longer issue: https://github.com/sile-typesetter/sile/issues/2006#issuecomment-1989668069  = "The way SILE implements obeying to lines in such environments is not perfect (it doesn't honor blank lines, which are skipped)." -- And mentioned opening a bug issue but apparently I wholly forgot ^^

Without looking to the code right now, it seems I was onto something (empty lines are skipped somewhere). Digging in the tests, we do have a test "Empty line are skipped" for #479 = PR #1766 "Conservative approach, skipping empty lines and lines containing only discardable nodes (incl. glues)."

But the latter (which went into 0.14.9 did fix some other tricky issue... So perhaps something is missing all along before event that fix, for the specific case of verbatim...

**Comment by Omikhleia** [comment]
Hmm... (again)... So #479 might have caused a side effect here maybe.
But digging further, we already had some weird thing on blank lines in verbatim environment: https://github.com/Omikhleia/omikhleia-sile-packages/blob/c3aa22153efcd98321e559cf9a791952e7909d06/packages/experimental/codehighlighter.lua#L40 = That was in 0.12.5 and visibly I had troubles there too.

**Comment by Omikhleia** [comment]
Hint: If we can have a MWE, we should track this debug: `SU.debug("typesetter", "Skipping a line containing only discardable nodes")` -- maybe the seenNonDiscardable condition should not apply as-is in cases where we want to honor empty lines (= typically verbatim)

**Comment by Omikhleia** [comment]
But then we'll maybe wonder "how can we know?" and notice all we have is a non-semantic `typesetter.parseppattern` setting... It does ring another bell on what are paragraphs (see #2111 § "Thoughts on paragraphing")...

**Comment by Omikhleia** [comment]
( Unrelated note FWIW: the _other_ issue reported by chrisvest on Mastodon regarding the split footnote is likely a bug too. Not the fact that the footnote gets split, -- but the fact that _on this very page_ there's visibly far enough space for it to fit _without_ needing the split if frames are broken as they are here -- or (rather) that much more more text could have been pushed to the main frame then., and the split would have been more legit.... And my deep feeling is that #1828 or a friend of it might be at play, with "gaps" sometimes occurring, although I _nearly_ never observed them in my own books, but on one occasion...)

**Comment by alerque** [comment]
The change in #1766 for #479 was first released in v0.14.9. I tried throwing an MWE against v0.14.8 and it behaved the same way, so I am still confused.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
